docs/source/crawler/basics.rst

```diff
@@ -27,5 +27,22 @@ a modified dictionary as its output once.
    or modified.  To help with creation of a new shallow copy with some fields
    adjusted, use :func:`~datalad.utils.updated`.
 
-Pipeline
---------
\ No newline at end of file
+Pipelines
+---------
+
+A pipeline is a series of functions gathered into a list. Each function take the
+output of its predecessor as its own input. The first function in the pipeline
+would need to be provided with input. The simplest pipeline could look like
+
+>>> def pipeline():
+...     [
+...     crawl_url('http://map.org/datasets'),
+...     a_href_match(".*.mat"),
+...     annex
+...     ]
+
+which crawls in a website, which is provided as input to the first function, then
+matches to all files that end in `.mat`, and those files are lastly inputted to
+the annex functions which simply annexes them. There can also be subpiplines within
+a pipeline, which are also denoted by `[]`. Two subpipelines that exist on top of
```

Comment:
describe more about generators... that each one of those could yield multiple times... if nothing is yielded -- pipeline effectively stopped.  There is also an exception `FinishPipeline` (challenge -- try to figure out how in sphinx to reference to the API, so it would make a hyperlink (don't do manually url) to http://docs.datalad.org/en/latest/generated/datalad.crawler.pipeline.html#datalad.crawler.pipeline.FinishPipeline


for subpipelines create a new section